src/main/java/com/spotify/autoscaler/db/BigtableCluster.java

```diff
@@ -61,6 +61,8 @@
 
   Optional<ErrorCode> errorCode();
 
+  Optional<Integer> overriddenMinNodes();
```

Comment:
My only concern about this whole change is having this two methods `overriddenMinNodes()` and `effectiveMinNodes()`. The other way was to have the logic somewhat in the database layer. @balaji you could check as well, do you have a better idea?